src/Behat/Gherkin/Parser.php

```diff
@@ -565,7 +566,16 @@ protected function parseExamples()
      */
     protected function parseTable()
     {
-        return new TableNode($this->parseTableRows());
+        try {
+            return new TableNode($this->parseTableRows());
+        }
+        catch(NodeException $e) {
+            throw new ParserException(
+                $e->getMessage() . $this->file ? ' in file '.$this->file : '',
```

Comment:
you are missing braces around the ternary. The current code is executed as `($e->getMessage() . $this->file) ? ' in file '.$this->file : ''` which is not what you want.